flowmachine/flowmachine/features/subscriber/contact_balance.py

```diff
@@ -136,14 +142,86 @@ def _make_query(self):
           FROM unioned as U) AS U
         JOIN total_events AS T
             ON U.subscriber = T.subscriber
-        GROUP BY U.subscriber, 
+        GROUP BY U.subscriber,
                  U.msisdn_counterpart,
                  T.events
         ORDER BY proportion DESC
         """
 
         return sql
 
+    def counterparts_subset(self, include_subscribers=False):
+        """
+        Returns the subset of counterparts. In some cases, we are interested in
+        obtaining information about the subset of subscribers contacts.
+
+        This method also allows one to get the subset of counterparts together
+        with subscribers by turning the `include_subscribers` flag to `True`.
+
+        Parameters
+        ----------
+        include_subscribers: bool, default True
+            Wether to include the list of subscribers in the subset as well.
+        """
+
+        return _ContactBalanceSubset(
+            contact_balance=self, include_subscribers=include_subscribers
+        )
+
+
+class _ContactBalanceSubset(SubscriberFeature):
```

Comment:
I'm not quite getting this one I think. Doesn't it just give you the set of people who had an inbound call/text?

Reply:
Ahh. I guess it would also allow you to get the set of subscribers who had an inbound call/text from some other subset of subscribers?

Why not make it a regular class rather than a hidden one?

Reply:
This class is meant to get the subset of subscriber contacts, given that not all contacts are reciprocals.

`Flowmachine` does not allow one to subset by the `msisdn` and `msisdn_counterpart` we can only subset by one or the other column. However, as in this case we want to calculate home location for the subset of our target subscribers and for all his contacts as well. This is basically what you mentioned in your previous comment.

I am indifferent between making this a regular or hidden class. My reasoning was that one can only get the subset of contacts by instantiating a `ContactBalance` class. I feel like getting the subset of subscribers would be a method of this class. So we would never want to instantiate a `ContactBalanceSubset` class on its own except when invoking this particular method.

I let you make the final decision on this one.